WordPress/openverse-frontend
# Toggle `SearchGridFilter` completely according to view width

## Fixes


Fixes #438 by @krysal
Related to #422 by @krysal 

## Description


This PR solves the problem of having a duplicated `SearchGridFilter` component and avoid repeating the ids of its subcomponents. See also #445.

## Testing Instructions

1. Run the project and search for something to go to the search result page.
2. Use the inspector of Developer Tools and observe a unique block of code for `SearchGridFilter`.
3. Extra: On the same search page, right-click on any checkbox of the filter sidebar and on "Inspect Accessibility Properties", each `checkbutton` should have its text set. This is the manual test I used to confirm it.

<img width="284" alt="CleanShot 2021-11-19 at 17 23 54@2x" src="https://user-images.githubusercontent.com/9145885/142693573-5a59b8e9-26b9-4e35-868d-e9af736ea3f8.png">

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
